test/test_testing.py

```diff
@@ -166,6 +179,7 @@ def test_can_connect_begin_with_initial_hooks(self):
         self.addCleanup(harness.cleanup)
 
         harness.begin_with_initial_hooks()
+        self.assertEqual(dict(pebble_ready_calls), {'foo': 1, 'bar': 1})
```

Comment:
I think we also want to assert that you 'can_connect' to foo and bar at this point as a side effect of pebble ready being called. Or is that already tested elsewhere?

Reply:
True -- done.